### PR TITLE
Split Service interfaces into multiple smaller ones. 

### PR DESCRIPTION
**Pull request by antoncarlgren**
Slight temporary refactoring of CompanyService, as it is the only one with an update method with a return value. Should be re-implemented once services have been refactored as a whole. #165

### CODE REVIEW

File: src/GT.Core/Services/Impl/GTCompanyService.cs [hunk 1]
```diff
@@ -144,7 +144,7 @@ public async Task<bool> ExistsByNameAsync(string name)
 		{
 			try
 			{
-				return await _companyRepository.GetAll().Where(e => e.Name == name).AnyAsync();
+				return await _companyRepository.GetAll().AnyAsync(e => e.Name == name);
```
Comment (JohanSaisa) [inline src/GT.Core/Services/Impl/GTCompanyService.cs:147]: Why not just return entity?

File: src/GT.Core/Services/Impl/GTCompanyService.cs [hunk 2]
```diff
@@ -218,14 +218,15 @@ public async Task<CompanyDTO> GetByIdAsync(string id)
...
 		}
 
-		public async Task<bool> UpdateAsync(CompanyDTO dto, string id)
+		
+		// TODO: Re-implement bool return after IGTService refactoring
+		public async Task UpdateAsync(CompanyDTO dto, string id)
 		{
 			try
 			{
 				if (dto.Id != id)
```
Comment (JohanSaisa) [inline src/GT.Core/Services/Impl/GTCompanyService.cs:227]: Shouldn't Id check happen before try catch?

File: src/GT.Core/Services/Impl/GTExperienceLevelService.cs [hunk 3]
```diff
@@ -116,5 +116,20 @@ public async Task<bool> ExistsByNameAsync(string name)
 				return null;
 			}
 		}
+
```
Comment (JohanSaisa) [inline src/GT.Core/Services/Impl/GTExperienceLevelService.cs:119]: Write TODO if we should implement interface?

File: src/GT.Core/Services/Impl/GTExperienceLevelService.cs [hunk 4]
```diff
@@ -116,5 +116,20 @@ public async Task<bool> ExistsByNameAsync(string name)
...
+		public Task<ExperienceLevelDTO?> GetByIdAsync(string id)
+		{
+			throw new NotImplementedException();
+		}
+
```
Comment (JohanSaisa) [inline src/GT.Core/Services/Impl/GTExperienceLevelService.cs:124]: Same as above.

File: src/GT.Core/Services/Impl/GTExperienceLevelService.cs [hunk 5]
```diff
@@ -116,5 +116,20 @@ public async Task<bool> ExistsByNameAsync(string name)
...
+		public Task UpdateAsync(ExperienceLevelDTO dto, string id)
+		{
+			throw new NotImplementedException();
+		}
+
```
Comment (JohanSaisa) [inline src/GT.Core/Services/Impl/GTExperienceLevelService.cs:129]: Same as above.

File: src/GT.Core/Services/Impl/GTLocationService.cs [hunk 6]
```diff
@@ -66,6 +66,16 @@ public async Task<LocationDTO> AddAsync(LocationDTO dto)
 			}
 		}
```
Comment (JohanSaisa) [inline src/GT.Core/Services/Impl/GTLocationService.cs:68]: TODO if we should implement?

File: src/GT.Core/Services/Impl/GTLocationService.cs [hunk 7]
```diff
@@ -66,6 +66,16 @@ public async Task<LocationDTO> AddAsync(LocationDTO dto)
...
 
+		public Task UpdateAsync(LocationDTO dto, string id)
+		{
+			throw new NotImplementedException();
+		}
+
```
Comment (JohanSaisa) [inline src/GT.Core/Services/Impl/GTLocationService.cs:73]: TODO if we should implement?

File: src/GT.Core/Services/Impl/GTLocationService.cs [hunk 8]
```diff
@@ -102,5 +112,11 @@ public async Task<bool> ExistsByNameAsync(string name)
 				return null;
 			}
 		}
+
```
Comment (JohanSaisa) [inline src/GT.Core/Services/Impl/GTLocationService.cs:115]: TODO if we should implement?

**Review by JohanSaisa — COMMENTED** [review]
Please see comments.

**Review by RasmusBroborg — CHANGES_REQUESTED** [review]
Review build errors before merge with main.

**Comment by antoncarlgren** [comment]
Closing PR/branch since issue requires more refactoring/work than previously thought, and should arguably be pushed back to a later date.

**Pull request closed without merging**